contracts/OgvStaking.sol

```diff
@@ -13,10 +17,10 @@ import {RewardsSource} from "./RewardsSource.sol";
 ///
 /// The balance received for staking (and thus the voting power and rewards
 /// distribution) goes up exponentially by the end of the staked period.
-contract OgvStaking is ERC20Votes {
+contract OgvStaking is Initializable, OwnableUpgradeable, UUPSUpgradeable, ERC20VotesUpgradeable {
     // 1. Core Storage
```

Comment:
We should allocate a slot gap here, right?

Reply:
ERC20VotesUpgradable has a big gap in ContextUpgradeable but there are a few initializable flags stored before that so I think 1 slot will collide with this before the gap is hit https://github.com/OriginProtocol/ousd-governance/blob/c894a651edf21aff2944be3eb2aaa94d374fa167/contracts/VoteLockerCurve.sol#L47 but it depends how things are packed.

From looking at the old VoteLockerCurve implementation we need a gap of 10 storage slots.

```
string 6 1
string 40 2
uint8 8 3
uint256 constant 0
uint256 constant 0
mapping 32 4
mapping 32 5
[] 32 6
uint256 32 7
mapping 32 8
mapping 32 9
address 10
```

We could jam a blank contract with a gap in before ERC20VotesUpgradeable to skip all the written storage slots from VoteLockerCurve since the first three inherited contracts are now identical.

I might be miscalculating here. I'd love to get a visual view of the storage but don't know of a tool/haven't had time to generate one.

Reply:
We can see the slots written during the first initialize here: 

https://dashboard.tenderly.co/tx/mainnet/0xc1493d425538b5fed552477235a8179d299af178905cf7e34111e5ae3790fa26/state-diff


|hex|slot|Notes|
|--|--|--|
|0x33|51|
|0xc9|201| New is mapping, should be no conflict|
|0xca|202| New is mapping, should be no conflict|
|0xcb|203|New totalySupply, set to decimals **CONFLICT**|
|0xce|206|In gap|
|0xd2|210|In gap|